sdk/nodejs/cmd/run-policy-pack/run.ts

```diff
@@ -162,17 +163,26 @@ export function run(opts: RunOpts): Promise<Record<string, any> | undefined> | P
     // that the "root" of the project is the cwd, if there's a tsconfig.json file here. Otherwise,
     // just tell ts-node to not load project options at all. This helps with cases like
     // pulumi/pulumi#1772.
-    const skipProject = !fs.existsSync("tsconfig.json");
+    let compilerOptions: object;
+    try {
+        const tsConfigPath = "tsconfig.json";
+        const tsConfigString = fs.readFileSync(tsConfigPath).toString();
+        const tsConfig = parseConfigFileTextToJson(tsConfigPath, tsConfigString).config;
+        compilerOptions = tsConfig["compilerOptions"] ?? {};
+    } catch (e) {
+        compilerOptions = {};
+    }
 
     if (opts.typeScript) {
         tsnode.register({
             typeCheck: true,
-            skipProject: skipProject,
+            skipProject: true,
             compilerOptions: {
                 target: "es6",
```

Comment:
Note that since the default Pulumi templates use `target: "es2016"`, this PR will lead to a change of beaviour for ~all users, where their tsnode compilation starts targeting es2016.  I believe that should be fine (and indeed likely good).  But it is a change, so we should be sure to do some further testing of this (for example, a run of the `examples` repo) to see if there are any issues raised.

Also related to https://github.com/pulumi/templates/issues/116 which would involve bumping up the default target for new projects, which after this change would (correctly, I believe) cause those changes to get picked up by the tsnode compilation.